### DISCUSSION TITLE
[@types/node] "undici-types" library has a security vulnerability and needs to be updated or replaced.

### DISCUSSION THREAD

**Discussion by Shirley-Ji-59** [DefinitelyTyped/DefinitelyTyped#68972] [Issues with a @types package] [answered]
I use @types/node in my project and then use Blackduck to scan it, and it reports a security vulnerability on undici-types 5.26.5, and the recommendation is upgrade to undici-types 6.7.1.
<img width="821" alt="Capture2" src="https://github.com/DefinitelyTyped/DefinitelyTyped/assets/78013718/02a019c1-3d7f-4a35-8f15-29d16293328f">

The reference is types/node/package.json
Could you upgrade it?

[a comment by an automated account is not shown]

**Comment by mcollina** [comment]
The vulnerability is on `undici`, not `undici-types`. This is also shown in your screenshot.
I think your scanner has a bug.

**Comment by Shirley-Ji-59** [comment]
![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/78013718/e7d2ba22-378e-4fdf-a66b-c9bfcea976a6)

From the Npm path, it appears that undici-types is also included within undici.


And from the Blackduck scan, you can also see the vulnerability is for undici-types/5.26.5
![Untitled](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/78013718/ef574ab2-4281-40a1-b3f2-2354965740ba)

**Reply by mcollina** [reply, marked as answer]
Both modules are created from the same repository. But _your tool is incorrect in its analysis_, as it's taking a vulnerability that is __clearly marked__ for the `undici`  package and applying it to `undici-types`. There is _literally no fix for this_, because it's __your tool problem__.


Note that the blackduck pages tells you that they did this assignment themselves, and did not follow the NVD record. Report this to them, it's a false positive on their side. Feel free to give them my contact.

**Reply by Shirley-Ji-59** [reply]
Thanks

**Reply by MA5TERHANZ** [reply]
Hi @Shirley-Ji-59, my name is Antoine, I work over here at Synopsys supporting Black Duck.
We came across this thread and spent some time yesterday doing some project restructuring on our OSS Knowledge-base.

We have since fixed this issue and Undici-types project should be reporting the proper vulnerabilities now (currently there are none reported to 5.26.5 as of making this comment).

Apologies for the confusion, please do not hesitate to reach out to us regarding any reports or confirming any BDSA findings data @ https://community.synopsys.com/s/

**Reply by Shirley-Ji-59** [reply]
@MA5TERHANZ  Thanks for your quick response. I think this is a nice discussion. I will close it.
